### PR TITLE
Split spec into page-per-section

### PR DESCRIPTION
**Pull request by illicitonion**

### CODE REVIEW

File: specification/modules/typing_notifications.rst [hunk 1]
```diff
@@ -5,10 +5,8 @@ Typing Notifications
 
 Users may wish to be informed when another user is typing in a room. This can be
 achieved using typing notifications. These are ephemeral events scoped to a
-``room_id``. This means they do not form part of the `Event Graph`_ but still
-have a ``room_id`` key.
-
-.. _Event Graph: `sect:event-graph`_
+``room_id``. This means they do not form part of the
+`Event Graph <index.html#event-graphs>`_ but still have a ``room_id`` key.
```
Comment (Kegsay) [inline specification/modules/typing_notifications.rst]: I would rather not hard-code a link back where possible. Can we do this any other way?

File: scripts/gendoc.py [hunk 2]
```diff
@@ -414,7 +433,7 @@ def main(target_name, keep_intermediates):
         help="Do not delete intermediate files. They will be found in tmp/"
     )
     parser.add_argument(
-        "--target", "-t", default="main",
+        "--target", "-t", default="all",
```
Comment (Kegsay) [inline scripts/gendoc.py]: If we're treating `all` as a special thing, we should document it in the help at the very least.

Comment (illicitonion) [inline scripts/gendoc.py]: Done

**Comment by Kegsay** [comment]
LGTM though I'm finding the Go is getting harder for me to grok as it continues to add more stuff.

This is mostly because my Go-fu is lacking, so I'm unable to spot things like off-by-ones / should you be using `defer` here, etc. If you can document your intent a bit more and/or reduce the scope of things I need to grok, I think I can get by. Given `main.go` is now >500 lines, it might be worth splitting things up so I don't need to CTRL+F for each variable mutation to see what else it touches (since it's never immediately clear from the diff if what you're assigning to is a global like `styledSpecCache` or a local/function-param-which-will-be-returned/etc when you're doing modifications on existing functions.

**Comment by illicitonion** [comment]
I will split up speculator in a separate PR :)

**Pull request merged**